DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -176,6 +185,14 @@ bool rememberMe
                 IssuedUtc = DateTime.UtcNow
             };
             HttpContext.SignInAsync("Identity.Application", new ClaimsPrincipal(claimsIdentity), authProperties);
+            if (!string.IsNullOrEmpty(returnUrl))
+            {
+                if (Url.IsLocalUrl(returnUrl))
+                {
+                    return Redirect(returnUrl);
+                }
+                logger.LogWarning($"Attempted login redirect to non-local returnUrl {returnUrl}");
+            }
             return RedirectToAction("Index", "Home");
```

Comment:
Could we separate this return url handling into a private method that returns an optional redirect. Then the return statement in this method can be something like:
`return RedirectToReturnUrl(returnUrl) ?? RedirectToAction("Index", "Home");`